independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/descriptor/loader/json/DirectoryResourceLoader.java

```diff
@@ -18,16 +17,16 @@ public DirectoryResourceLoader(Path dir) {
     }
 
     @Override
-    public <T> T loadResource(String name, ResourceInputStreamConsumer<T> consumer) throws IOException {
-        Path resource = dir.resolve(name);
-        if (!Files.exists(resource)) {
-            throw new IOException("Failed to locate " + resource);
+    public <T> T loadResourcePath(String name, ResourcePathConsumer<T> consumer) throws IOException {
+        Path path;
+        if (name == null || name.isEmpty()) {
+            path = dir;
```

Comment:
This is strange. `name` should not never be null or empty IMHO

Reply:
I think I did that to be able to load the root directory (for file listing). I don't really remember why I added it, and since the codestart directory is never the root, we should be fine removing it..